src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,507 @@
+        return 1
+
+    return parsed
+
+
+def split_pdf_to_chunks(pdf_content: bytes, pages_per_chunk: int) -> List[Dict[str, Any]]:
+    """
+    Split a PDF into multi-page chunks using pypdfium2.
+
+    Returns a list of dictionaries containing the chunk bytes and page range metadata.
+    Note: this currently buffers each chunk in-memory; consider streaming in future upgrades.
+    """
+
+    chunks: List[Dict[str, Any]] = []
+
+    if pages_per_chunk <= 0:
+        pages_per_chunk = 1
+
+    pdf = pdfium.PdfDocument(pdf_content)
+    total_pages = len(pdf)
+
+    try:
+        for chunk_index, start_zero in enumerate(range(0, total_pages, pages_per_chunk)):
+            end_zero = min(start_zero + pages_per_chunk, total_pages)
+            page_indices = list(range(start_zero, end_zero))
+
+            new_pdf = pdfium.PdfDocument.new()
+            try:
+                new_pdf.import_pages(pdf, page_indices)
+
+                buffer = BytesIO()
+                try:
+                    new_pdf.save(buffer)
+                    chunk_bytes = buffer.getvalue()
+                finally:
+                    buffer.close()
+            finally:
+                new_pdf.close()
+
+            start_page = start_zero + 1
+            end_page = end_zero
+            chunk_info: Dict[str, Any] = {
+                "bytes": chunk_bytes,
+                "chunk_index": chunk_index,
+                "start_page": start_page,
+                "end_page": end_page,
+                "page_count": end_page - start_page + 1,
+            }
+            chunks.append(chunk_info)
+
+    finally:
+        pdf.close()
+
+    return chunks
+
+
+def get_pdf_page_count(pdf_content: bytes) -> int:
+    """Get the number of pages in a PDF using pypdfium2."""
+    try:
+        pdf = pdfium.PdfDocument(pdf_content)
+        page_count = len(pdf)
+        pdf.close()
+        return page_count
+    except Exception as e:
+        logger.warning(f"Failed to get PDF page count: {e}")
+        return 1  # Assume single page on error
+
+
+# POST /v2/submit_job
+@router.post(
+    "/submit_job",
+    responses={
+        200: {"description": "Jobs were successfully submitted"},
+        500: {"description": "Error encountered while submitting jobs."},
+        503: {"description": "Service unavailable."},
+    },
+    tags=["Ingestion"],
+    summary="submit jobs to the core nv ingestion service for processing with PDF splitting",
+    operation_id="submit_job_v2",
+)
+async def submit_job_v2(
+    request: Request, response: Response, job_spec: MessageWrapper, ingest_service: INGEST_SERVICE_T
+):
+    with tracer.start_as_current_span("http-submit-job-v2") as span:
+        try:
+            # Add custom attributes to the span
+            span.set_attribute("http.method", request.method)
+            span.set_attribute("http.url", str(request.url))
+            span.add_event("Submitting file for processing (V2)")
```

Comment:
Re: Julio's comment on monolithic functions. 

We could probably make this a modular helper so its more declarative.